designer/client/page-edit.js

```diff
@@ -118,15 +123,50 @@ class PageEdit extends React.Component {
       })
   }
 
+  onChangeTitle = e => {
+    const input = e.target
+    const title = input.value
+    this.setState({
+      title: title,
+      generatedPath: this.generatePath(title)
+    })
+  }
+
+  onChangePath = e => {
+    const input = e.target
+    const path = input.value.startsWith('/') ? input.value : `/${input.value}`
+    const sanitizedPath = path.replace(/\s/g, '-')
+    this.setState({
+      path: sanitizedPath
+    })
+  }
+
+  generatePath (title) {
+    let path = toUrl(title)
+    const { data, page } = this.props
+
+    let count = 1
+    while (data.findPage(path) && page.title !== title) {
+      if (count > 1) {
+        path = path.substr(0, path.length - 2)
+      }
+      path = `${path}-${count}`
+      count++
+    }
+    return path
+  }
+
   render () {
     const { data, page } = this.props
     const { sections } = data
+    const { title, path, generatedPath, controller, section } = this.state
 
```

Comment:
possibly worth assigning `path || generatedPath || page.path` to a variable, same with controller || page.controller || '' for legibility

Reply:
Fixed